src/Repository/DBALMessageRepository.php

```diff
@@ -48,13 +49,16 @@ protected function executeQuery(QueryBuilder $queryBuilder)
 
         try {
             @$stmt->execute($bind);
-        } catch (DriverException $exception) {
+        } catch (DriverException | DeadlockException $exception) {
 
             // Only keep SQLState HY000 with ErrorCode 2006 (MySQL server has gone away)
```

Comment:
Maybe update the comment to specify what is the code 40001 to be more explicit :)

Reply:
Yes sure